prymer/primer3/primer3.py

```diff
@@ -310,21 +311,21 @@ def _screen_pair_results(
             valid: bool = True
             if (
                 primer_pair.left_primer.longest_dinucleotide_run_length
-                > design_input.primer_and_amplicon_params.primer_max_dinuc_bases
+                > design_input.primer_max_dinuc_bases
             ):  # if the left primer has too many dinucleotide bases, fail it
                 dinuc_pair_failures.append(primer_pair.left_primer)
                 valid = False
             if (
                 primer_pair.right_primer.longest_dinucleotide_run_length
-                > design_input.primer_and_amplicon_params.primer_max_dinuc_bases
+                > design_input.primer_max_dinuc_bases
             ):  # if the right primer has too many dinucleotide bases, fail it
                 dinuc_pair_failures.append(primer_pair.right_primer)
                 valid = False
             if valid:  # if neither failed, append the pair to a list of valid designs
                 valid_primer_pair_designs.append(primer_pair)
         return valid_primer_pair_designs, dinuc_pair_failures
 
-    def design(self, design_input: Primer3Input) -> Primer3Result:  # noqa: C901
+    def design(self, design_input: Primer3Parameters) -> Primer3Result:  # noqa: C901
```

Comment:
I think this would need to take:

```python
amplicon_parameters: AmpliconParameters | None, probe_parameter: ProbeParameters | None
```

What you have works for the current set of use cases, but one primer3 task we haven't implemented yet supports designing primer pairs and probes concurrently, and then you need both sets of parameters.

Reply:
I leave it for now, and we can decide later if we want to have two parameters to `design`, or to create a third parameters class:
```python
class AmpliconAndProbeParameters(Primer3Parameters):
    amplicon_params: AmpliconParams
    probe_params: ProbeParams
```